### PR TITLE
finished get topics

### PR DESCRIPTION
**Pull request by Jay-R-Mac**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const request = require("supertest");
+const app = require("../app.js");
+const connection = require("../db/connection.js");
+const seed = require("../db/seeds/seed.js");
+const data = require("../db/data/test-data/index.js");
+const { string } = require("pg-format");
```
Comment (elijaharley) [inline __tests__/app.test.js]: This might be an auto import but can be removed - we don't need pg-format in the test file.

File: README.md [hunk 2]
```diff
@@ -1,3 +1 @@
-# Northcoders News API
-
-For instructions, please head over to [L2C NC News](https://l2c.northcoders.com/courses/be/nc-news).
+Two .env files must be created to use this repo a development and test .env with a PGdatabase and PGdatabase test respectively
```
Comment (elijaharley) [inline README.md]: Be explicit about this - .env.test and .env.development and PGDATABASE - this is important as typos will prevent the connection to the db from working . Don't forget to advise them to npm i dotenv

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        body.forEach((topic)=>{
+          expect(topic).toMatchObject({
+            slug : expect.any(String),
+            description: expect.any(String)
+          })
+        })
+        
+      });
+  });
+});
```
Comment (elijaharley) [inline __tests__/app.test.js:39]: Nice work on this test.

File: app.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const app = express();
+
+app.get("/api/topics", sendTopics)
+
+
+
+
+
+
+
```
Comment (elijaharley) [inline app.js]: We have an error handler for all errors that is being tested "/api/notvalid" however there is currently no error handler for this. Please add an app.all error handler with a path of '*' and a status of 404 for any invalid requests. Please also add a 500 error handler for server errors.

File: model/model.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const db = require("../db/connection");
+
+function getTopics() {
+  return db.query("SELECT * FROM topics;").then(({ rows }) => {
+    return rows;
+  });
+}
+
```
Comment (elijaharley) [inline model/model.js:8]: Nice destructuring here.

**Review by elijaharley — CHANGES_REQUESTED** [review]
There is a test currently for a 404 error that is not being handled by an error handler. I have left notes inline detailing this. The README also needs updating. You're very close with this ticket now.

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+          });
+        });
+      });
+  });
+});
+
+describe("error handling", () => {
+  it("should return a 404 error status code and a message when using a GET", () => {
+    return request(app)
+      .get("/notvalid")
```
Comment (pippaamy) [inline __tests__/app.test.js:44]: great job!

File: app.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const express = require("express");
+const { sendTopics } = require("./controller/controller")
+
```
Comment (pippaamy) [inline app.js]: im confused is there two app.js files here?

File: db/app.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const { sendTopics } = require("./controller/controller");
+
```
Comment (pippaamy) [inline db/app.js:3]: & here?

File: db/app.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+const app = express();
+
+app.get("/api/topics", sendTopics);
+
+app.all("/*", (req, res) => {
```
Comment (pippaamy) [inline db/app.js:8]: good error handling though

**Review by pippaamy — COMMENTED** [review]
good job, lets just get rid of one of those app.js files then all good to merge :)

**Pull request merged**